crates/type-system/src/ontology/data_type/wasm.rs

```diff
@@ -0,0 +1,24 @@
+#![allow(
+    non_snake_case,
+    reason = "We want camelCase types in TS, and this module is WASM only"
```

Comment:
```suggestion
    reason = "We want camelCase functions in TS, and this module is WASM only"
```

Reply:
This feels weird though, can't we rename the function in the WASM attribute?

Reply:
I'm not just talking about functions, I'm on about variable names, etc. The main problem is arguments _within_ functions, we can rename a function but we can't rename the arguments within them. So renaming
```rust
pub fn is_valid_data_type(data_type_obj: &IDataType)
```

will result in 
```typescript
function isValidDataType(data_type_obj: DataType)
```

rather than 
```typescript
function isValidDataType(dataTypeObj: DataType)
```

Reply:
Gotcha, makes sense!

Reply:
I guess those aren't quite types.
```suggestion
    reason = "We want camelCase variable names in TS, and this module is WASM only"
```